Files/DataModels/SidebarPinnedModel.cs.bak

```diff
@@ -0,0 +1,430 @@
+                    MainPage.SideBarItems.Add(favoriteSection);
+                }
+
+                if (App.AppSettings.ShowLibrarySwitch)
+                {
+                    if (!MainPage.SideBarItems.Contains(librarySection))
+                    {
+                        MainPage.SideBarItems.Add(librarySection);
+                    }                        
+                }
+
+                MainPage.SideBarItems.EndBulkOperation();
+            }
+            finally
+            {
+                MainPage.SideBarItemsSemaphore.Release();
+            }
+
+            await ShowHideRecycleBinItemAsync(App.AppSettings.PinRecycleBinToSideBar);
+        }
+
+        /// <summary>
+        /// Removes stale items in the navigation sidebar
+        /// </summary>
+        public void RemoveStaleSidebarItems()
+        {
+            // Remove unpinned items from sidebar
+            for (int i = 0; i < favoriteSection.ChildItems.Count(); i++)
+            {
+                if (favoriteSection.ChildItems[i] is LocationItem)
+                {
+                    var item = favoriteSection.ChildItems[i] as LocationItem;
+                    if (!item.IsDefaultLocation && !Items.Contains(item.Path))
+                    {
+                        favoriteSection.ChildItems.RemoveAt(i);
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets the icon for the items in the navigation sidebar
+        /// </summary>
+        /// <param name="path">The path in the sidebar</param>
+        /// <returns>The icon code</returns>
+        public string GetItemIcon(string path)
+        {
+            string iconCode;
+            
+            if (path.Equals(AppSettings.DesktopPath, StringComparison.OrdinalIgnoreCase))
+            {
+                iconCode = "\ue9f1";
+            }
+            else if (path.Equals(AppSettings.DownloadsPath, StringComparison.OrdinalIgnoreCase))
+            {
+                iconCode = "\uE91c";
+            }
+            else if (path.Equals(AppSettings.DocumentsPath, StringComparison.OrdinalIgnoreCase))
+            {
+                iconCode = "\uea11";
+            }
+            else if (path.Equals(AppSettings.PicturesPath, StringComparison.OrdinalIgnoreCase))
+            {
+                iconCode = "\uea83";
+            }
+            else if (path.Equals(AppSettings.MusicPath, StringComparison.OrdinalIgnoreCase))
+            {
+                iconCode = "\uead4";
+            }
+            else if (path.Equals(AppSettings.VideosPath, StringComparison.OrdinalIgnoreCase))
+            {
+                iconCode = "\uec0d";
+            }
+            else if (Path.GetPathRoot(path).Equals(path, StringComparison.OrdinalIgnoreCase))
+            {
+                iconCode = "\ueb8b";
+            }
+            else
+            {
+                iconCode = "\uea55";
+            }
+
+            return iconCode;
+        }
+    }
+}
```

Comment:
Is this file necessary?

Reply:
no, sorry i used WinMerge and forget to remove that file

Reply:
@oleitao No problem, just make sure to remove it before we merge the PR.